src/opm/output/eclipse/Summary.cpp

```diff
@@ -134,105 +136,86 @@ namespace {
+            makeEntities('W', Opm::EclIO::SummaryNode::Category::Well, well_name);
+            makeExtraEntities(extra_well_vectors, Opm::EclIO::SummaryNode::Category::Well, well_name);
         }
 
         for (const auto& grp_name : sched.groupNames()) {
             if (grp_name != "FIELD") {
-                makeEntities('G', SN::Category::Group, grp_name);
-
-                entities.emplace_back("GMCTP", SN::Category::Group, ::Opm::Location());
-                entities.back().namedEntity(grp_name)
-                .parameterType(SN::Type::Mode);
-
-                entities.emplace_back("GMCTW", SN::Category::Group, ::Opm::Location());
-                entities.back().namedEntity(grp_name)
-                .parameterType(SN::Type::Mode);
-
-                entities.emplace_back("GMCTG", SN::Category::Group, ::Opm::Location());
-                entities.back().namedEntity(grp_name)
-                .parameterType(SN::Type::Mode);
+                makeEntities('G', Opm::EclIO::SummaryNode::Category::Group, grp_name);
+                makeExtraEntities(extra_group_vectors, Opm::EclIO::SummaryNode::Category::Group, grp_name);
             }
         }
 
-        makeEntities('F', SN::Category::Field, "FIELD");
-
-        entities.emplace_back("FMCTP", SN::Category::Field, ::Opm::Location());
-        entities.back().namedEntity("FIELD")
-        .parameterType(SN::Type::Mode);
-
-        entities.emplace_back("FMCTW", SN::Category::Field, ::Opm::Location());
-        entities.back().namedEntity("FIELD")
-        .parameterType(SN::Type::Mode);
-
-        entities.emplace_back("FMCTG", SN::Category::Field, ::Opm::Location());
-        entities.back().namedEntity("FIELD")
-        .parameterType(SN::Type::Mode);
+        makeEntities('F', Opm::EclIO::SummaryNode::Category::Field, "FIELD");
+        makeExtraEntities(extra_field_vectors, Opm::EclIO::SummaryNode::Category::Field, "FIELD");
 
         return entities;
     }
 
-    std::vector<Opm::SummaryConfigNode>
+    std::vector<Opm::EclIO::SummaryNode>
     requiredSegmentVectors(const ::Opm::Schedule& sched)
     {
-        using SN = Opm::SummaryConfigNode;
-        auto ret = std::vector<SN>{};
-
-        auto sofr = SN{ "SOFR", SN::Category::Segment, ::Opm::Location() }
-            .parameterType(SN::Type::Rate);
-
-        auto sgfr = SN{ "SGFR", SN::Category::Segment, ::Opm::Location() }
-            .parameterType(SN::Type::Rate);
-
-        auto swfr = SN{ "SWFR", SN::Category::Segment, ::Opm::Location() }
-            .parameterType(SN::Type::Rate);
-
-        auto spr = SN{ "SPR", SN::Category::Segment, ::Opm::Location() }
-            .parameterType(SN::Type::Pressure);
+        std::vector<Opm::EclIO::SummaryNode> ret {};
+
+        static const Opm::EclIO::SummaryNode::Category category { Opm::EclIO::SummaryNode::Category::Segment };
+        static const std::vector<std::pair<std::string,Opm::EclIO::SummaryNode::Type>> requiredVectors {
```

Comment:
Please remove `static` here.  The `requiredSegmentVectors()` function is only called during the initial configuration step.